src/LibraryUtilities.ts

```diff
@@ -175,22 +190,52 @@ export function constructLibraryItem(
         // current 'result' as the parent node for child nodes to be appended.
         // 
         let parentNode = inclusive ? null : result;
+        let nodeFound: boolean = false;
+        let isLeafNode: boolean = true;
 
         for (let j = 0; j < typeListNodes.length; j++) {
-
             let fullyQualifiedName = typeListNodes[j].fullyQualifiedName;
-            if (!fullyQualifiedName.startsWith(includePath)) {
-                continue; // Not matching, skip to the next type node.
+            let fullyQualifiedNameParts = fullyQualifiedName.split('.');
+            let lastIncludedPart = includeParts[includeParts.length - 1];
+
+            // Check if a matching node is found, and if the fullyQualifiedName contains
+            // the (whole) last word in includePath. 
+            // E.g. This ensures EllipseArc nodes to be excluded in the Ellipse category.
+            if (fullyQualifiedName.startsWith(includePath) && _.contains(fullyQualifiedNameParts, lastIncludedPart)) {
```

Comment:
We can't use `_.contains` here, e.g. the following scenario:

`[ "DSCore", "Math", "Randomizer", "Random" ]` will be included when it should not have been.